src/app/shell/Header.tsx

```diff
@@ -338,6 +339,9 @@ class Header extends React.PureComponent<Props, State> {
     }
 
     this.updateVendorEngrams(this.props.account || undefined);
+
+    const vendorEngramDropActive = this.props.vendorDrops.some(isDroppingHigh);
```

Comment:
Put this in your mapStateToProps function. That way, you'll only re-render when `vendorEngramDropActive` changes, not when any of vendorDrops changes. Plus, your component will depend on the minimum state.